web-common/src/components/button-toggle-group/SubButton.svelte

```diff
@@ -0,0 +1,72 @@
+<script lang="ts">
+  import { getContext } from "svelte";
+  import {
+    ButtonGroupContext,
+    buttonGroupContext,
+  } from "./ButtonToggleGroup.svelte";
+
+  import Tooltip from "../tooltip/Tooltip.svelte";
+  import TooltipContent from "../tooltip/TooltipContent.svelte";
+
+  export let key: number | string;
+  export let tootips: {
+    selected?: string;
+    unselected?: string;
+    disabled?: string;
+  } = undefined;
+
+  const {
+    registerSubButton,
+    toggleSubButton,
+    selectedKey,
+    firstKey,
+    lastKey,
+    disabledKeys,
+  }: ButtonGroupContext = getContext(buttonGroupContext);
+
+  registerSubButton(key);
+
+  $: isDisabled = disabledKeys.includes(key);
+  $: isSelected = $selectedKey === key;
+
+  const baseStyles = `shrink flex flex-row items-center px-1 py-1
+  transition-transform duration-100`;
+
+  $: textStyle = isDisabled
+    ? "text-gray-400"
+    : "text-gray-700 hover:text-gray-900 ";
+
+  $: bgStyle = isDisabled
+    ? "bg-white"
+    : isSelected
+    ? "bg-gray-100 hover:bg-gray-200 "
+    : "bg-white hover:bg-gray-50 ";
```

Comment:
Not important for this PR, but a side note: We have this pattern many places in our codebase where we need to add different class names based on a variety of conditions. I wonder if its worth adding a simple `classnames()` utility for this, where you could rewrite the above (and similar scenarios) as:
```javascript
bgStyle = classnames({
  "bg-white": isDisabled,
  "bg-gray-100 hover:bg-gray-200": !isDisabled && isSelected,
  "bg-white hover:bg-gray-50": !isDisabled && !isSelected
})
```

Reply:
I'm torn on this. I can see how that might be useful, but it's also another layer of abstraction beyond plain JS string manipulation -- if I see a function like that, I'll always have to wonder "hmmm what is that function doing?" and go look at it's code to make sure it doesn't do anything surprising, vs conditionals over strings being totally explicit in-line.